### PR TITLE
Added performance test for ChangeBinOffset.

### PR DESCRIPTION
**Pull request by SimonHeybrock**
This is a preparation for upcoming refactoring of this algorithm.

No related issue.
No release notes.

Code review only.

### CODE REVIEW

**Comment by FedeMPouzols** [comment]
This adds a performance test with a couple of cases for an algorithm before refactoring it. The performance test is simple and seemingly fast enough that I don't see any problem in including it. That's a good practice, so :shipit:

File: Framework/Algorithms/test/ChangeBinOffsetTest.h [hunk 1]
```diff
@@ -164,4 +166,43 @@ class ChangeBinOffsetTest : public CxxTest::TestSuite {
...
+    AnalysisDataService::Instance().clear();
+    delete suite;
+  }
+
+  ChangeBinOffsetTestPerformance() {
+    auto input = WorkspaceCreationHelper::Create2DWorkspaceBinned(10000, 1000);
+    AnalysisDataService::Instance().add("input", input);
+    auto inputEvent =
+        WorkspaceCreationHelper::CreateEventWorkspace(10000, 1000, 5000);
+    AnalysisDataService::Instance().add("inputEvent", inputEvent);
```
Comment (FedeMPouzols) [inline Framework/Algorithms/test/ChangeBinOffsetTest.h]: @SimonHeybrock : just curious. I see that you intentionally put the workspaces in the ADS and then take the inputs from there. Have you observed any performance implications when doing that?

Comment (SimonHeybrock) [inline Framework/Algorithms/test/ChangeBinOffsetTest.h]: @FedeMPouzols I did not want to include creation of the workspaces in the part that is timed. I suppose I could have done the same by storing it in a member variable?

Comment (FedeMPouzols) [inline Framework/Algorithms/test/ChangeBinOffsetTest.h]: Ah I see. That makes perfect sense. Yes, that's one possibility if you don't want to use the ADS.

**Pull request merged**